### PR TITLE
Add `extract_from_folder` function

### LINKED ISSUE #53: Add `extract_from_folder` function that takes in a path (str) to a folder/directory
An `extract_from_folder` function can be added to the package that takes in a path as a string to a folder/directory. It gets all files with a `.png` file extension inside the folder and extracts the data.

### PR DESCRIPTION
**Pull request by gabb222**
Fixes #53

### CODE REVIEW

File: tspng/extraction.py [hunk 1]
```diff
@@ -110,3 +112,34 @@ def extract_from_files(paths: List[str], mime_type: str=MIME_TYPE) -> Dict:
     for path in paths:
         nested_dict[path] = extract_from_file(path, mime_type)
     return nested_dict
+
+def extract_from_folder(path: str, mime_type: str=MIME_TYPE) -> Dict:
+    '''
+    Returns a nested dictionary of metadata from a folder of TSPNG file paths.
+
+        Parameters:
+                path (list[str]): List of file paths
```
Comment (volks73) [inline tspng/extraction.py:121]: This should be just `str` and "A path to a folder".

File: tspng/extraction.py [hunk 2]
```diff
@@ -110,3 +112,34 @@ def extract_from_files(paths: List[str], mime_type: str=MIME_TYPE) -> Dict:
...
+        Returns:
+                (dict): Dictionary containing metadata of each file
+
+        Raises:
+                Exception: If path is not a directory
+                Exception: If path does not contain a PNG file
+    '''
+    #check if directory
+    if not os.path.isdir(path):
+        raise Exception(f"{path} is not to a directory.")
```
Comment (volks73) [inline tspng/extraction.py:134]: I would add a "The" to the error message, i.e., "The {path} is not to a directory.".

File: tspng/extraction.py [hunk 3]
```diff
@@ -110,3 +112,34 @@ def extract_from_files(paths: List[str], mime_type: str=MIME_TYPE) -> Dict:
...
+    # return all files as a list
+    file_list = []
+    for file in os.listdir(path):
+        # check the files which are end with specific extension
+        if file.endswith(".png"):
```
Comment (volks73) [inline tspng/extraction.py:139]: There are some weird edge cases out there when working with files, extensions, and file names. I would use the [`os.path.splitext`](https://docs.python.org/3/library/os.path.html#os.path.splitext) function to get the file extension and check if it is a `.png` to automatically handle any of those weird edge cases.

File: tspng/extraction.py [hunk 4]
```diff
@@ -110,3 +112,34 @@ def extract_from_files(paths: List[str], mime_type: str=MIME_TYPE) -> Dict:
...
+            # print path name of selected files
+            file_list.append(os.path.join(path, file))
+    #check if any PNG files in directory
+    if file_list == []:
+        raise Exception(f"{path} does not contain a PNG file.")
```
Comment (volks73) [inline tspng/extraction.py:144]: I would change the message to "The {path} does not contain any PNG files.".

Comment (volks73) [reply tspng/extraction.py:144]: I do like this check. It will be very helpful to users when things go wrong. Nice touch.

**Review by volks73 — APPROVED** [review]
Looks good.

Some minor changes to investigate, but I don't want to block if possible.

File: tspng/extraction.py (hunk 2 above)
Comment (gabb222) [reply tspng/extraction.py:134]: Change made as requested

File: tspng/extraction.py (hunk 3 above)
Comment (gabb222) [reply tspng/extraction.py:139]: Change made as requested

File: tspng/extraction.py (hunk 4 above)
Comment (gabb222) [reply tspng/extraction.py:144]: Thanks, change made as requested

**Pull request merged**